src/main/java/com/guciowons/shoppingbasket/Basket/BasketService.java

```diff
@@ -1,67 +1,71 @@
 package com.guciowons.shoppingbasket.Basket;
 
 import com.guciowons.shoppingbasket.Exception.*;
+import com.guciowons.shoppingbasket.Product.Product;
 import com.guciowons.shoppingbasket.Product.ProductService;
-import feign.FeignException;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 public class BasketService {
     private final BasketSummarizer basketSummarizer;
-    private final BasketDao basketDao;
+    private final BasketRepository basketRepository;
     private final ProductService productService;
 
-    public BasketService(BasketSummarizer basketSummarizer, BasketDao basketDao, ProductService productService) {
+    public BasketService(BasketSummarizer basketSummarizer, BasketRepository basketRepository, ProductService productService) {
         this.basketSummarizer = basketSummarizer;
-        this.basketDao = basketDao;
+        this.basketRepository = basketRepository;
         this.productService = productService;
     }
 
     public Basket createBasket() {
-        Basket newBasket = new Basket(basketDao.getAll().size());
-        basketDao.save(newBasket);
+        Basket newBasket = new Basket(basketRepository.findAll().size());
```

Comment:
Zobacz jakie powinno być ID przy tworzeniu obiektów przez zapisem do bazy. ID powinien nadawać system bazodanowy, a nie Ty sam.

Tym bardziej tutaj, jak masz już obsługę wielu koszyków, to ten sposób nadawania ID może powodować powstawanie koszyków o tym samym ID:
1. Tworzysz koszyk, dostajesz ID = 0
2. Tworzysz koszyk, dostajesz ID = 1
3. Usuwasz koszyk z ID = 0
4. Tworzysz koszyk, znowu dostajesz ID = 1 !!!